### PR TITLE
Remove react test renderer from I18n Specs

### PR DESCRIPTION
**Pull request by SeanArmstrong**
# Description
React test renderer was just used for shallow rendering of I18n specs. These specs have been refactored to use enzyme/snapshot testing

Snapshot tests need to be triple checked before merging.

NOTE: Module has been re-added as it is required by enzyme - https://github.com/airbnb/enzyme/issues/875

### CODE REVIEW

**Comment by andrewjtait** [comment]
Fixes https://github.com/Sage/carbon/issues/1445

File: src/components/i18n/__spec__.js [hunk 1]
```diff
@@ -10,85 +9,52 @@ import { elementsTagTest, rootTagTest } from '../../utils/helpers/tags/tags-spec
...
-      it('when component is rendered with only a key', () => {
-
-        shallowRenderer.render(
-          <I18n scope='foo' />
-        );
-
-        const output = shallowRenderer.getRenderOutput();
-        expect(output.type).toEqual('span');
-        expect(output.props.children).toEqual('bar');
+      describe('when component is rendered with only a key', () => {
```
Comment (abramin) [inline src/components/i18n/__spec__.js]: Should this be `scope`? Can't see anything called `key`

File: src/components/i18n/__spec__.js [hunk 2]
```diff
@@ -10,85 +9,52 @@ import { elementsTagTest, rootTagTest } from '../../utils/helpers/tags/tags-spec
...
-
-        shallowRenderer.render(
-          <I18n scope='foo' inline={ false } />
-        );
-
-        const output = shallowRenderer.getRenderOutput();
-        expect(output.type).toEqual('div');
-        expect(output.props.children).toEqual('bar');
+      describe('when component is rendered with inline set to false', () => {
+        it('renders the value of the key in a div', () => {
```
Comment (abramin) [inline src/components/i18n/__spec__.js]: scope?

**Comment by SeanArmstrong** [comment]
This seems to be adding 
```
    console.warn node_modules/react-addons-test-utils/index.js:33
      Warning: ReactTestUtils has been moved to react-dom/test-utils. Update references to remove this warning.
    console.error node_modules/fbjs/lib/warning.js:35
      Warning: Shallow renderer has been moved to react-test-renderer/shallow. Update references to remove this warning.
```

File: src/components/i18n/__spec__.js [hunk 3]
```diff
@@ -10,85 +9,52 @@ import { elementsTagTest, rootTagTest } from '../../utils/helpers/tags/tags-spec
...
-          <I18n scope='foo' markdown={ true }/>
-        );
-
-        const output = shallowRenderer.getRenderOutput();
-        expect(output.type).toEqual('span');
-        expect(output.props.dangerouslySetInnerHTML).toEqual({
-          __html: 'something <strong>bold</strong>'
+      describe('when component is rendered inline', () => {
+        it('renders the parsed markdown in a span', () => {
+          spyOn(i18n, 't').and.returnValue('something __bold__');
```
Comment (ianoxley) [inline src/components/i18n/__spec__.js]: Prefer `jest.fn()`

File: src/components/i18n/__spec__.js [hunk 4]
```diff
@@ -10,85 +9,52 @@ import { elementsTagTest, rootTagTest } from '../../utils/helpers/tags/tags-spec
...
-          <I18n scope='foo' markdown={ true } inline={ false } />
-        );
-
-        const output = shallowRenderer.getRenderOutput();
-        expect(output.type).toEqual('div');
-        expect(output.props.dangerouslySetInnerHTML).toEqual({
-          __html: '<p>something <strong>bold</strong></p>\n'
+      describe('when component is rendered as a block', () => {
+        it('renders the parsed markdown in a div', () => {
+          spyOn(i18n, 't').and.returnValue('something __bold__');
```
Comment (ianoxley) [inline src/components/i18n/__spec__.js]: Prefer `jest.fn()`

File: src/components/i18n/__spec__.js [hunk 5]
```diff
@@ -10,85 +9,52 @@ import { elementsTagTest, rootTagTest } from '../../utils/helpers/tags/tags-spec
...
-          <I18n scope='foo' markdown={ true } />
-        );
-
-        const output = shallowRenderer.getRenderOutput();
-        expect(output.type).toEqual('span');
-        expect(output.props.dangerouslySetInnerHTML).toEqual({
-          __html: 'some &lt;span&gt;html&lt;/span&gt;'
+      describe('when component is rendered with html in the value', () => {
+        it('parses the html correctly', () => {
+          spyOn(i18n, 't').and.returnValue('some <span>html</span>');
```
Comment (ianoxley) [inline src/components/i18n/__spec__.js]: Prefer `jest.fn()`

File: package-lock.json [hunk 6]
```diff
@@ -9317,16 +9317,6 @@
...
-    "react-test-renderer": {
-      "version": "15.6.1",
-      "resolved": "https://registry.npmjs.org/react-test-renderer/-/react-test-renderer-15.6.1.tgz",
-      "integrity": "sha1-Am9KW7VVJmH9LMS7zQ1LyKNev34=",
-      "dev": true,
-      "requires": {
-        "fbjs": "0.8.15",
-        "object-assign": "4.1.1"
-      }
-    },
```
Comment (andrewjtait) [inline package-lock.json]: should this be re-added?

**Pull request closed without merging**